src/credentials.cpp

```diff
@@ -214,33 +216,37 @@ namespace preferences
 	}
 }
 
+const int AES_KEY_SIZE = 256;
+
 // TODO: Key-stretching (bcrypt was recommended)
 std::string build_key(const std::string& server, const std::string& login)
 {
 	std::ostringstream out;
 	out << login << get_system_username() << server;
-	return out.str();
-}
-
-// FIXME: XOR encryption is a really terrible choice - swap it out for something better!
-// TODO: Maybe use cryptopp or something for AES encryption?
-static std::string xor_crypt(std::string text, const std::string& key)
-{
-	const size_t m = key.size();
-	for(size_t i = 0; i < text.size(); i++) {
-		text[i] ^= key[i % m];
+	while(out.tellp() < AES_KEY_SIZE / 8) {
+		out.put('x');
 	}
-	return text;
+	return out.str();
 }
 
 std::string encrypt(const std::string& text, const std::string& key)
 {
-	return xor_crypt(text, key);
+	std::shared_ptr<unsigned char> out(new unsigned char[text.size() + 1], std::default_delete<unsigned char[]>());
+	out.get()[text.size()] = 0;
+	AES_KEY ak;
+	AES_set_encrypt_key(reinterpret_cast<const unsigned char*>(key.c_str()), AES_KEY_SIZE, &ak);
+	AES_encrypt(reinterpret_cast<const unsigned char*>(text.c_str()), out.get(), &ak);
+	return reinterpret_cast<const char*>(out.get());
 }
 
 std::string decrypt(const std::string& text, const std::string& key)
```

Comment:
The `text` parameter and the return value aren't really text. They are binary data. Same for the `encrypt()` function.

It would be best to pass the encrypted and unencrypted data as `std::vector<unsigned char>`. That would also eliminate the `reinterpret_cast`s which you currently need.